app/components/tag_list_input_component/tag_list_input_component.html.haml

```diff
@@ -0,0 +1,20 @@
+- #%div{ "data-controller": "tag-list-input-component--tag-list-input" }
+%div{ "data-controller": "tag-list-input-component--tag-list-input", "data-tag-list-input-component--tag-list-input-highlight-class-value": "changed" }
```

Comment:
The controller name seems unnecessarily repeated. I'm guessing that's due to `export default` in the JS. 
It would be worth explicitly specifying the class name to give it a shorter name like `TagListInputController`. (Actually I'd advocate for always specifying the classname at the top of the file anyway.)

Bonus points if you rename with even less words ;) maybe something like `TagSelectController`?

Reply:
I believe that's stimulus limitation, but I'll have another go at this. Part of the issue is the component prefix is added automatically ( in this case `tag-list-input-component`).

Reply:
There is actually a way to register a controller with a custom name ie : https://stimulus.hotwired.dev/reference/controllers#registering-controllers-manually

But that goes against convention over configuration, given that stimulus controllers in `app/webpacker/controllers` and controller in `app/components/**/` are loaded automatically. I understand the name is a bit unpalatable but I think we should stick to following the convention.

Reply:
I see, thanks for looking into it. I vaguely recall trying to find a better solution for arranging JS in components back at the beginning, but there wasn't one..